pandas/tests/io/json/test_pandas.py

```diff
@@ -4,11 +4,13 @@
 from pandas.compat import (range, lrange, StringIO,
                            OrderedDict, is_platform_32bit)
 import os
+import boto3
 
 import numpy as np
 from pandas import (Series, DataFrame, DatetimeIndex, Timestamp,
                     read_json, compat)
 from datetime import timedelta
+from moto import mock_s3
```

Comment:
This will probably have to be an `importorskip` so that it runs without mock_s3.

```python
moto = pytest.importorskip("moto")
```

then use `@moto.mock_s3` below.

Reply:
Fixed.

Just out of curiosity, why do you want these tests to be skipped in a given scenario? Wouldn't you just want to ensure that the dep. was included in all the right places and always run?

Reply:
We don't install the optional dependencies in every environment, so we need to skip tests that would otherwise error with an ImportError on an optional dependency.